src/composables/useApiClient.ts

```diff
@@ -0,0 +1,118 @@
+import { ref, reactive } from 'vue'
+import { useCookies } from '@vueuse/integrations/useCookies'
+import { useJwt } from '@vueuse/integrations/useJwt'
+import type { JwtPayload } from 'jwt-decode'
+import type { JSONResponse } from '@models/JSONSchema'
+import LogService from '@services/LogService'
+import { CatClient, type CancelablePromise, type AuthPermission, type AuthResource } from 'ccat-api'
+
+
+
+const { DEV } = import.meta.env
+
+const getPort = () => {
+    if (DEV) return 1865
+    if (window.location.port == '443' || window.location.port == '80') return undefined
+    return parseInt(window.location.port)
+}
+
+
+/**
+ * A function that wraps the promise request into a try/catch block
+ * and check for errors to throw to the UI
+ * @param request The axios promise function to await
+ * @param success The message to return in case of success
+ * @param error The message to return in case of error
+ * @param log The log message/array of stuff to show
+ * @returns A JSONResponse object containing status, message and optionally a data property
+*/
+const tryRequest = async <T>(
```

Comment:
take it back to `api.ts` because it is used also outside of vue reactivity. also remember to export it.